### DISCUSSION TITLE
Google user profile claims not showing up in the JWT

### DISCUSSION THREAD

**Discussion by agusterodin** [nextauthjs/next-auth#3014] [Help] [answered]
How do I get a JWT from Google that contains the user profile claims (name, picture, etc)?

I am aware that this library performs a separate call to Google (with https://www.googleapis.com/oauth2/v1/tokeninfo?alt=json) after getting the id token to fetch the profile for the purpose of front-end consumption. The issue with this is that I need the claims to be in the bearer token since I pass the JWT token directly to a backend server and it uses profile claims (such as name and picture) directly. Yes, I added the profile scope to the next-auth Google provider object.

React-Google-Login ([react-google-login](https://github.com/anthonyjgrove/react-google-login)) which I migrated from provided me with a token that contained all profile claims. This may be due to the client JS Google library being entirely different.

Here are some resources that may be relevant. None of them were able to help me figure it out unfortunately.

https://stackoverflow.com/questions/32175174/is-it-possible-to-get-profile-information-in-an-id-token-from-google
https://github.com/nextauthjs/next-auth/issues/393

Thanks!

**Comment by balazsorban44** [comment, marked as answer]
The v4 Google provider will use the OIDC well known configuration and thus return you the ID token.

See the code here https://github.com/nextauthjs/next-auth/blob/beta/src/providers/google.ts

keep in mind that you should use the access token for API requests, not the ID token.

**Reply by agusterodin** [reply]
Appreciate it ❤️

**Reply by balazsorban44** [reply]
v4 is in public beta, feel free to give it a try!

Here is the migration guide:
https://next-auth.js.org/getting-started/upgrade-v4

(_and by the way, thank you 💖_)

**Reply by agusterodin** [reply]
Ps: Everyone has their own style but I would consider enabling one-time donations. That is more stomachable for many people.

**Reply by balazsorban44** [reply]
We do have custom donations 🙂: https://opencollective.com/nextauth
